#7528: add new ethernet microbenchmark, cleanup and re-enable others

SeanNijjar opened the pull request:
Adds new ping latency microbenchmark which measures the time taken (in cycles) for an erisc to send a message (of user specifyable size) to the core on the other end of the ethernet link and receive a response.

Other attributes that can be specified are the number of samples to measure as well as the number of concurrent channels to send pings over (where each channel is a separate, concurrent transaction/packet on the same link).

Initial results indicate an almost exactly 1us roundtrip ping time for a flit (16B payload + 16B message available tail - not a true flit but very close)

Bidirectional bandwidth and ring latency microbenchmarks are also re-enabled and improved.

Ring latency improves to about 655ns per hop for an 8 hop ring.

Banwidth is improved to around 23.5GBps for packet size range from 4k to 16k per packet.

Some samples. Buffer size in bytes is on the left. Number of channels/buffers instantiated per direction are on the right. So a num channels of 2 indicates 4 channels total (2 in sender direction and 2 in receiver).

| Buf Size (B)   | # Channels |
|-------|----|
| 36864 | 2  |
| 32768 | 2  |
| 24576 | 3  |
| 16384 | 4  |
| 8192  | 9  |
| 4096  | 18 |
| 2048  | 30 |
| 1024  | 30 |

### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/7528)

### Problem description/ What's changed
Adds more ubench and re-enables/fixes others

Describe the approach used to solve the problem.
N/A

### Checklist
- [ ] Post commit CI passes
  - post_commit: https://github.com/tenstorrent/tt-metal/actions/runs/9966462855
  - t3000 frequent: https://github.com/tenstorrent/tt-metal/actions/runs/9963963712
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

ttmchiou committed:
#0: add t3k ethernet tests

ttmchiou commented on tests/scripts/t3000/run_t3000_frequent_tests.sh:
```diff
@@ -124,7 +125,7 @@ run_t3000_falcon40b_tests() {
 run_t3000_tests() {
   # Run ethernet tests
-  #run_t3000_ethernet_tests
+  run_t3000_ethernet_tests
```
https://github.com/tenstorrent/tt-metal/pull/10026
will need this PR/branch.
CI invokes tests via invoking the bash function itself.

SeanNijjar replied:
Tried with this commit cherry picked and it worked.

ttmchiou commented on .github/workflows/t3000-frequent-tests.yaml:
```diff
@@ -19,6 +19,8 @@ jobs:
         test-group: [
           { name: "t3k tteager tests", arch: wormhole_b0, cmd: run_t3000_tteager_tests, timeout: 60, 
           runs-on: [arch-wormhole_b0, "config-t3000", "in-service", "runner-test", "bare-metal", "pipeline-functional"], owner_id: ULMEPM2MA}, #Sean Nijjar
+          { name: "t3k tteager tests", arch: wormhole_b0, cmd: run_t3000_ethernet_tests, timeout: 60, 
```
good to merge after this name change,
no need to rerun workflow

Commit:
#7528: add new ethernet microbenchmark, cleanup and re-enable others

Adds new ping latency microbenchmark which measures the time taken (in
cycles) for an erisc to send a message (of user specifyable size) to the
core on the other end of the ethernet link and receive a response.

Other attributes that can be specified are the number of samples to
measure as well as the number of concurrent channels to send pings over
(where each channel is a separate, concurrent transaction/packet on the
same link).

Initial results indicate an almost exactly 1us roundtrip ping time for a
flit (16B payload + 16B message available tail - not a true flit but
very close)

Bidirectional bandwidth and ring latency microbenchmarks are also
re-enabled and improved.

Ring latency improves to about 655ns per hop for an 8 hop ring.

Banwidth is improved to around 23.5GBps for packet size range from 4k to
16k per packet.

Some samples. Buffer size in bytes is on the left. Number of
channels/buffers instantiated per direction are on the right.
So a num channels of 2 indicates 4 channels total (2 in sender direction
and 2 in receiver).

| Buf   |    |
| Size  | #  |
| (B)   | Ch |
|-------|----|
| 36864 | 2  |
| 32768 | 2  |
| 24576 | 3  |
| 16384 | 4  |
| 8192  | 9  |
| 4096  | 18 |
| 2048  | 30 |
| 1024  | 30 |
